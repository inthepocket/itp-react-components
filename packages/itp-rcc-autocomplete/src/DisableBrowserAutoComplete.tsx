import * as React from 'react';

const style = {
  display: 'none',
};

/**
 * Disables browser autocomplete
 * https://gist.github.com/niksumeiko/360164708c3b326bd1c8
 */
const DisableBrowserAutoComplete = ({
  className = 'disable-browser-autocomplete',
  children = null,
}) => (
  <form autoComplete="off" className={className}>
    <input autoComplete="false" name="hidden" style={style} type="text" />
    {children}
  </form>
);

interface DisableBrowserAutoCompleteProps {
  className?: string;
  children: React.ReactNode;
}

export default DisableBrowserAutoComplete;
