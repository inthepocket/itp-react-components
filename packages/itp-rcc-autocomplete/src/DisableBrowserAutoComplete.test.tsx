import * as React from 'react';
import { create } from 'react-test-renderer';
import DisableBrowserAutoComplete from './DisableBrowserAutoComplete';

test('DisableBrowserAutoComplete snapshot', () => {
  const tree = create((
    <DisableBrowserAutoComplete>
      <p>Browser autocomplete disabled</p>
    </DisableBrowserAutoComplete>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});