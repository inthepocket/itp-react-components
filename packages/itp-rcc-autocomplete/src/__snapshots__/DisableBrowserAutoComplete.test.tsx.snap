// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisableBrowserAutoComplete snapshot 1`] = `
<form
  autoComplete="off"
  className="disable-browser-autocomplete"
>
  <input
    autoComplete="false"
    name="hidden"
    style={
      Object {
        "display": "none",
      }
    }
    type="text"
  />
  <p>
    Browser autocomplete disabled
  </p>
</form>
`;
